python-checks/src/main/java/org/sonar/python/checks/NumpyIsNanCheck.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.BinaryExpression;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.HasSymbol;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.plugins.python.api.tree.Tree;
+
+@Rule(key = "S6725")
+public class NumpyIsNanCheck extends PythonSubscriptionCheck {
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.COMPARISON, ctx -> checkForIsNan(ctx));
+  }
+
+  private static void checkForIsNan(SubscriptionContext ctx) {
+    BinaryExpression be = (BinaryExpression) ctx.syntaxNode();
+    String value = be.operator().value();
+    if (!("==".equals(value) || "!=".equals(value))) {
+      return;
+    }
+    checkOperand(ctx, be.leftOperand(), be);
+    checkOperand(ctx, be.rightOperand(), be);
+  }
+
+  private static void checkOperand(SubscriptionContext ctx, Expression operand, BinaryExpression be) {
+    HasSymbol cast;
+    if (operand.is(Tree.Kind.QUALIFIED_EXPR)) {
+      cast = (QualifiedExpression) operand;
+    } else if (operand.is(Tree.Kind.NAME)) {
+      cast = (Name) operand;
+    } else {
+      return;
+    }
+    Symbol symbol = cast.symbol();
+    if (symbol != null && "numpy.nan".equals(symbol.fullyQualifiedName())) {
+      ctx.addIssue(be, "Equality checks should not be made against \"numpy.nan\". Use numpy.isnan() instead.");
```

Comment:
Small feedback: the issue message should be imperative. The description part is then explained in the "Why is this an issue" section.
I'd suggest to simply use `Don't perform an equality check against "numpy.nan".`
When implemented, the quick fix could then read `Use numpy.isnan`

Still, good job and kudos on your first merged PR!

Reply:
Addressed in 42964f1f0e484919e3c122e61ab1337e964bbfb8